src/database/UIDatabase.js

```diff
@@ -35,36 +33,53 @@ export class UIDatabase {
     this.database = database;
   }
 
+  ERRORS = {
```

Comment:
TODO: wonder if at some point it will be good to manage all `ERRORS` in a single source file and import rather than define in scope of each file?

Reply:
definitely. I didn't really like doing this but I do like returning more than a boolean